src/plugin/fieldViews/CheckboxFieldView.ts

```diff
@@ -1,9 +1,19 @@
 import type { Node } from "prosemirror-model";
 import type { EditorView } from "prosemirror-view";
 import { AttributeFieldView } from "./AttributeFieldView";
+import type { BaseFieldSpec } from "./FieldView";
 
 export type CheckboxFields = { value: boolean };
 
+export interface CheckboxField extends BaseFieldSpec<CheckboxFields> {
```

Comment:
We have a type `CheckboxFields` and interface `CheckboxField` which is a bit confusing. I think `CheckboxFields` could be renamed to something more appropriate. Possibly how Fields is used in `AttributeFieldView` more accurately represents values?

Reply:
Yes, let's rename `CheckboxField`, it should really be `CheckboxValue`. We could do with doing that across the board as you say, the original class names that type parameter `NodeValue`! Shall we include that in this PR?

Reply:
I think we could change `CheckboxFields` -> `CheckboxValue` in this PR as we are adding a new interface called `CheckboxField`. Other changes could reasonably be deferred to a separate PR.